File: backend/app/schemas/support.py
from typing import List

from pydantic import BaseModel

from app.schemas.customer_accounts import AccountManagerSummary


class SupportTimelineEntry(BaseModel):
    id: int
    author_name: str
    author_type: str
    author_label: str
    entry_kind: str
    visibility: str
    is_customer_visible: bool
    message: str
    source_label: str = ""
    created_at: str


class SupportCaseSummary(BaseModel):
    id: int
    internal_case_id: int
    customer_account_id: int
    case_number: str
    subject: str
    severity: str
    status: str
    customer_status_label: str
    summary: str
    created_at: str
    updated_at: str
    assigned_team: str
    comments: List[SupportTimelineEntry]


class SupportCaseCreateRequest(BaseModel):
    subject: str
    severity: str
    summary: str


class SupportCaseCreateResponse(BaseModel):
    message: str
    case: SupportCaseSummary


class SupportCaseReplyRequest(BaseModel):
    message: str


class SupportCaseReplyResponse(BaseModel):
    message: str
    case: SupportCaseSummary


class SupportContactSummary(BaseModel):
    name: str
    title: str
    email: str
    phone: str
    hours: str
    urgent_message: str


class SupportSeverityGuide(BaseModel):
    level: str
    response_target: str
    description: str


class SupportQuickHelpArticle(BaseModel):
    id: int
    title: str
    description: str


class SupportResourcesResponse(BaseModel):
    account_manager: AccountManagerSummary
    support_contact: SupportContactSummary
    severity_guidance: List[SupportSeverityGuide]
    quick_help_articles: List[SupportQuickHelpArticle]

File: frontend/src/types/support.ts
export type SupportTimelineEntry = {
    id: number;
    author_name: string;
    author_type: string;
    visibility: string;
    is_customer_visible: boolean;
    message: string;
    created_at: string;
};

export type SupportCase = {
    id: number;
    internal_case_id: number;
    customer_account_id: number;
    case_number: string;
    subject: string;
    severity: string;
    status: string;
    summary: string;
    created_at: string;
    updated_at: string;
    assigned_team: string;
    comments: SupportTimelineEntry[];
};

export type SupportCaseCreateRequest = {
    subject: string;
    severity: string;
    summary: string;
};

export type SupportCaseCreateResponse = {
    message: string;
    case: SupportCase;
};

export type SupportResources = {
    account_manager: {
        name: string;
        title: string;
        email: string;
        phone: string;
    };
    support_contact: {
        name: string;
        title: string;
        email: string;
        phone: string;
        hours: string;
        urgent_message: string;
    };
    severity_guidance: {
        level: string;
        response_target: string;
        description: string;
    }[];
    quick_help_articles: {
        id: number;
        title: string;
        description: string;
    }[];
};

export type SupportCaseReplyRequest = {
    message: string;
};

export type SupportCaseReplyResponse = {
    message: string;
    case: SupportCase;
};